packages/editor/src/store/actions.js

```diff
@@ -177,15 +178,26 @@ export const savePost =
 				edits,
 				options
 			);
-		dispatch( { type: 'REQUEST_POST_UPDATE_FINISH', options } );
 
-		const error = registry
+		let error = registry
 			.select( coreStore )
 			.getLastEntitySaveError(
 				'postType',
 				previousRecord.type,
 				previousRecord.id
 			);
+
+		if ( ! error ) {
+			await applyFilters(
```

Comment:
This looks more like an "action" than a filter, the only reason it's a filter is because we want it to be async. This makes me wonder if we shouldn't just support "async actions" instead.

`await doAction( 'editor.SavePost' )`

Reply:
Yes, webpack also distinguishes between sync and async hooks, and it's a good thing to have in a JavaScript project.

We could introduce a new `doAsyncAction` API which is similar to `doAction` but awaits between calls and aborts on a rejected promise. The `addAction` API could stay the same. The entire difference is in how actions are called, not how they are registered.

If the caller makes a mistake and calls `doAction` instead of `doAsyncAction`, we could issue a warning in case an action callback returns a promise (or thenable in general) that would be ignored by `doAction`.

Reply:
curious why this needs to be async though, is the thought that `options` might get changed? if so pass that to the filter, if not just use action without async?

Reply:
It's async because saving post is in principle async. It involves a REST request for saving the post, and we wait for the response etc. And the callbacks registered with the `__unstableSavePost` hook typically want to do some additional async work related to saving. For example, the `edit-post` package registers a callback for saving metaboxes. I.e., in addition to the standard post saving, it will go through metabox forms on the page, and save their fields with a separate POST request to the server.